src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/service/ProductStockRequestApiServiceImpl.java

```diff
@@ -21,24 +21,254 @@
  * That means that one need to lookup
  * {@link org.eclipse.tractusx.puris.backend.stock.domain.model.ProductStock} and return it
- * according to the API speicfication.
+ * according to the API specification.
  */
 @Component
+@Slf4j
 public class ProductStockRequestApiServiceImpl implements RequestApiService {
 
+    private static final OkHttpClient CLIENT = new OkHttpClient();
+
+    @Autowired
+    private RequestService requestService;
+
+    @Autowired
+    private MaterialService materialService;
+
+    @Autowired
+    private ProductStockService productStockService;
+
+    @Autowired
+    private ProductStockSammMapper productStockSammMapper;
+
+    @Autowired
+    private EdcAdapterService edcAdapterService;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Value("${edc.dataplane.public.port}")
+    String dataPlanePort;
+
+    @Value("${edc.controlplane.host}")
+    String dataPlaneHost;
+
+    @Value("${edc.idsUrl}")
+    private String ownEdcIdsUrl;
+
+    @Value("${own.bpnl}")
+    private String ownBPNL;
+
+
+
+    public static <T> Predicate<T> distinctByKey(
+            Function<? super T, ?> keyExtractor) {
+
+        Map<Object, Boolean> seen = new ConcurrentHashMap<>();
+        return t -> seen.putIfAbsent(keyExtractor.apply(t), Boolean.TRUE) == null;
+    }
+
     @Override
-    public void handleRequest(Request request) {
-        throw new NotYetImplementedException("Implement Request Flow for " +
-                "ProductStockRequestApiService");
+    public void handleRequest(RequestDto requestDto) {
+
+        log.info(String.format("param requestDto %s", requestDto));
+
+        // quickfix: don´t persist request
+        requestDto.setState(DT_RequestStateEnum.WORKING);
+        /*
+        //request has been created on post
+        Request correspondingRequest = requestService.findByInternalUuid(requestDto.getUuid());
+
+        // as soon as we're working on it, we need to set the state to working.
+        log.info(String.format("found correspondingRequest %s", correspondingRequest));
+        correspondingRequest = requestService.updateState(correspondingRequest, DT_RequestStateEnum.WORKING);
+        */
+        String partnerIdsUrl = requestDto.getHeader().getSenderEdc();
+        //Partner requestingPartner = partnerService.findByBpnl();
+
+        // contains either productStockSamms or messageContentError
+        List<MessageContentDto> resultProductStocks = new ArrayList<>();
+
+        String requestingPartnerBpnl = requestDto.getHeader().getSender();
+
+        for (ProductStockRequestForMaterialDto productStockRequestDto : requestDto.getPayload()) {
+
+            //if (messageContentDto instanceof ProductStockRequestForMaterialDto) {
+
+            //ProductStockRequestForMaterialDto productStockRequestDto =
+            //        (ProductStockRequestForMaterialDto) messageContentDto;
+
+            // TODO determine data
```

Comment:
Please resolve todos in class and clean up (delete / incorporate) outcommented information.

Reply:
Done